### ISSUE TITLE
BodyDefferingAsyncHandler should not retry

### ISSUE THREAD

**Issue by jogoussard** [AsyncHttpClient/async-http-client#1555]
Hi
I found out that the BodyDefferingAsyncHandler does not play well with the default retries on GET.
If the connection is closed after the caller thread has started consuming the body, the default behavior of the client is to retry the request, pouring the new body into the same output stream as the first request.
Unless I'm mistaken BodyDefferingAsyncHandler should overwrite 'onRetry' and throw an exception there, unless there's a cleaner way to do this.
The current workaround is to configure the client with 0 retries, but as it is global, that's not a good solution.

**Comment by slandelle** [comment]
Contribution would be greatly appreciated.

**Comment by jogoussard** [comment]
I would but I cannot push the branch it seems.

```
git push origin fixes/issue_30
remote: Permission to AsyncHttpClient/async-http-client.git denied to jogoussard.
fatal: unable to access 'https://github.com/AsyncHttpClient/async-http-client.git/': The requested URL returned error: 403
```

**Comment by slandelle** [comment]
Of course you can't push on this repository!
You need to fork it and then open a [pull request](https://help.github.com/articles/about-pull-requests/).

**Comment by jogoussard** [comment]
Fork a complete repo under my account and then push a PR on this fork? Then you grab the changes? Sounds quite overkill to me - here's a patch.
[issue_30.patch.zip](https://github.com/AsyncHttpClient/async-http-client/files/2182044/issue_30.patch.zip)

**Comment by slandelle** [comment]
Please understand that pull requests are the more convenient for people who maintain open source projects (and who are usually swamped and do OSS on their free time):
* easier to review and provide feedback/request for changes
* easier to merge

So please go with the standard process.

**Comment by jogoussard** [comment]
https://github.com/AsyncHttpClient/async-http-client/pull/1560

**Comment by slandelle** [comment]
Closed by #1560
